libs/FileManagerKit/source/File.cpp

```diff
@@ -209,7 +209,7 @@ auto File::getSHA256() -> std::array<uint8_t, 32>
 	mbedtls_sha256_init(&context);
 	mbedtls_sha256_starts(&context, 0);
 
-	for (auto i = 0; i < size(); i += std::size(buffer)) {
+	for (unsigned i = 0; i < size(); i += std::size(buffer)) {
```

Comment:
What's the good use/benefit behind this change ?

Reply:
`auto i = 0` yields an `int`

`std::size()` returns an `size_t / unsigned int`

when doing the comparison, there was a warning 

```
/Users/ladislas/dev/leka/LekaOS/libs/FileManagerKit/source/File.cpp: In member function 'virtual std::array<unsigned char, 32> leka::FileManagerKit::File::getSHA256()':
/Users/ladislas/dev/leka/LekaOS/libs/FileManagerKit/source/File.cpp:212:21: warning: comparison of integer expressions of different signedness: 'int' and 'size_t' {aka 'unsigned int'} [-Wsign-compare]
  212 |  for (auto i = 0; i < size(); i += std::size(buffer)) {
      |                   ~~^~~~~~~~
```

it's nothing but it's still noise, and the compiler is always right. It may not have an impact now but it could in the future. Using `unsigned` tells the compiler and the reader that we know what we are doing.

Actually, I'll even use `size_t` to make it more explicit! I'll also take a look at the `size()` function, see if we can fix it there instead.

thanks for pointing out @HPezz 👍